src/components/HorizontalVideoCard/HorizontalVideoCard.js

```diff
@@ -1,15 +1,30 @@
 import React from "react";
-import { Link } from "react-router-dom";
+import { Link, useLocation } from "react-router-dom";
 import { deleteFromLike } from "../../services/deleteFromLikes";
+import { deleteFromWatchLater } from "../../services/deleteFromWatchLater";
 import { useAuth } from "../../context/AuthContext";
 import { useLike } from "../../context/LikeContext";
+import { useWatchLater } from "../../context/WatchLaterContext";
 import "./HorizontalVideoCard.css";
 
 export const HorizontalVideoCard = ({ video }) => {
 	const {
 		authState: { token },
 	} = useAuth();
 	const { likeDispatch } = useLike();
+
+	const { watchLaterDispatch } = useWatchLater();
+
+	const location = useLocation();
+
+	const deleteHandler = () => {
+		if (location.pathname === "/liked") {
+			deleteFromLike(video._id, token, likeDispatch);
+		}
+		if (location.pathname === "/watchlater") {
+			deleteFromWatchLater(video._id, token, watchLaterDispatch);
+		}
+	};
```

Comment:
Suggestion : you can pass delete handler of both like and watchlater prop to horizontal card , so you dont need to check location